types/tampermonkey/index.d.ts

```diff
@@ -529,5 +531,158 @@ declare function GM_notification(
  */
 declare function GM_setClipboard(
     data: string,
-    info?: string | { type?: string | undefined; mimetype?: string | undefined }
+    info?: Tampermonkey.ContentType,
 ): void;
+
+// GM.*
+
+/**
+ * `GM` has all the `GM_*` apis in promisified form
+ */
+declare const GM: Readonly<{
+    // Styles
+
+    /**
+     * Adds the given style to the document and returns the injected style element.
+     */
+    addStyle(css: string): Promise<HTMLStyleElement>;
+
+    // Storage
+
+    /** Sets the value of `name` to the storage */
+    setValue(name: string, value: any): Promise<void>;
+
+    /** Gets the value of 'name' from storage */
+    getValue<TValue>(name: string, defaultValue?: TValue): Promise<TValue>;
+
+    /** Deletes 'name' from storage */
+    deleteValue(name: string): Promise<void>;
+
+    /** Lists all names of the storage */
+    listValues(): Promise<string[]>;
+
+    /**
+     * Adds a change listener to the storage and returns the listener ID.
+     * The `remote` argument of the callback function shows whether this value was
+     * modified from the instance of another tab (`true`) or within this script
+     * instance (`false`). Therefore this functionality can be used by scripts of
+     * different browser tabs to communicate with each other.
+     * @param name Name of the observed variable
+     */
+    addValueChangeListener(name: string, listener: Tampermonkey.ValueChangeListener): Promise<number>;
```

Comment:
Should we make it generic so that `listener` returns typed `oldValue` and `newValue`?

Reply:
Leaving it as `any` has the advantage that you can do things like `(.., oldValue: number | string, newValue: string, ..) => {}`. Either way, I think it is out of the scope of this PR.